src/common/transformations/src/transformations/common_optimizations/mark_subgraphs_to_keep_in_mixed_precision.cpp

```diff
@@ -0,0 +1,313 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/mark_subgraphs_to_keep_in_mixed_precision.hpp"
+
+#include "itt.hpp"
+#include "openvino/op/util/broadcast_base.hpp"
+#include "openvino/op/util/gather_base.hpp"
+#include "openvino/opsets/opset10.hpp"
+#include "openvino/opsets/opset2.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/common_optimizations/mark_precision_sensitive_shapeof_subgraphs.hpp"
+#include "transformations/convert_precision.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "transformations/rt_info/reduceop_path.hpp"
+#include "transformations/utils/utils.hpp"
+
+using namespace std;
+
+namespace ov {
+namespace pass {
+
+MarkNormalizationOps::MarkNormalizationOps() {
+    MATCHER_SCOPE(MarkNormalizationOps);
+    auto ops_to_be_kept_fp32 = pattern::wrap_type<opset2::MVN, opset10::MVN, opset10::NormalizeL2>();
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto& node = m.get_match_root();
+        if (!node)
+            return false;
+
+        disable_fp16_compression(node);
+        return true;
+    };
+    auto m = make_shared<pattern::Matcher>(ops_to_be_kept_fp32, matcher_name);
+    register_matcher(m, callback);
+}
+
+std::shared_ptr<Node> propagate_through_ops =
+    pattern::wrap_type<opset10::Squeeze,
+                       opset10::Unsqueeze,
+                       opset10::Reshape,
+                       op::util::BroadcastBase,
+                       op::util::BinaryElementwiseArithmetic,
+                       op::util::UnaryElementwiseArithmetic,
+                       opset10::MVN,
+                       opset2::MVN,
+                       opset10::NormalizeL2,
+                       opset10::Sqrt,
+                       opset10::StridedSlice,
+                       opset10::ReduceSum,
+                       opset10::ReduceMean,
+                       opset10::Slice,
+                       opset10::VariadicSplit,
+                       opset10::Split,
+                       op::util::GatherBase,
+                       opset10::Concat,
+                       opset10::Convert,  // through Convert can go only to Constants
+                       opset10::Constant,
+                       opset10::Tile>();
+
+class PropagateUpMarkToKeepInMixedPrecision : public pass::MatcherPass {
```

Comment:
Could you add comments to at least propagate up and down transformations to explain what they do, why propagate both is required

Reply:
Added explanation why we need both markings